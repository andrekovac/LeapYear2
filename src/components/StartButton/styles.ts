import styled from "styled-components/native";

export const ButtonWide = styled.TouchableOpacity`
  width: 100%;
`;

type StyledTextProps = {
  size?: number;
  color?: string;
};

export const StyledText = styled.Text<StyledTextProps>`
  font-size: ${props => (props.size ? props.size + "px" : "30px")};
  font-weight: 200;
  text-align: center;
  color: ${props => props.color ?? "black"};
`;

export const ButtonTextWrapper = styled.View`
  padding: 10px 0;
  background-color: white;
  border-radius: 35px;
  align-items: center;
`;
