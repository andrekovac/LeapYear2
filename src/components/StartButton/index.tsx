import React, { FC } from "react";
import styled from "styled-components/native";

import StyledText from "../StyledText";
import { ButtonTextWrapper } from "./styles";


type StartButtonProps = {
  onPress: () => void;
}
const StartButton: FC<StartButtonProps> = ({ onPress, children }) => (
  <ButtonWide onPress={onPress}>
    <ButtonTextWrapper>
      <StyledText>{children}</StyledText>
    </ButtonTextWrapper>
  </ButtonWide>
);

const ButtonWide = styled.TouchableOpacity`
  width: 100%;
`;

export default StartButton;