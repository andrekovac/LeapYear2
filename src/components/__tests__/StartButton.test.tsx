import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import StartButton from "../StartButton";

describe("StyledText", () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <StartButton onPress={() => null}>Start Button Snapshot</StartButton>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders the correct button title", () => {
    const { queryByText } = render(
      <StartButton onPress={() => null}>My Button</StartButton>
    );
    expect(queryByText("My Button")).not.toBeNull();
  });
});
