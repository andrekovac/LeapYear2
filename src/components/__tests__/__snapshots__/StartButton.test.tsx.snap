// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 35,
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "200",
          "textAlign": "center",
        }
      }
    >
      Start Button Snapshot
    </Text>
  </View>
</View>
`;
